pkg/networkpath/metricsender/metric_sender.go

```diff
@@ -0,0 +1,12 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package metricsender holds the interface used to send Metrics with Agent Sender and Statsd sender
+package metricsender
```

Comment:
Just a nit on package name here. `metricsender.MetricSender` stutters, I think you could have something like `metric.Sender` instead which has less repetition and is clearer to read. This would also lead to `metric.agentSender`, etc. below, which is more readable.